Feature: update detector and SAM pipe region

rahmans1 opened the pull request:
Updated the detector and SAM pipe region in accordance with the latest CAD file received from Randy Wilson (JLAB). 

Main changes:

1) Introduced material definitions for Inconel625 and Inconel625(10%)-Air(90%) mixture
2) Updated radial dimensions and z-extents of detector pipe, neckdown, SAM pipe, bellows and flanges. Set bellows flanges (yellow) to Inconel 625 material and the bellows to Inconel-Air mixture(brown). Set everything in blue to Aluminum including detector pipe, neckdown and SAM pipe with flange. 
![ExperimentalGeometry](https://user-images.githubusercontent.com/7409132/185771843-68060ca0-d5da-438b-a0ac-5ff1ee78ce7f.PNG)
![ZoomedExperimentalGeometry](https://user-images.githubusercontent.com/7409132/185771844-7c90d85a-4c98-48dd-822a-8c1c7a73b18b.PNG)
3) Subtracted out a hole in hall logical volume and hall wall detector logical volume to avoid overlaps. 
4) Added an annulus of Aluminum as placeholders for the tubes holding the SAMs in place.
5) Moved the SAM placeholders to new location as per the CAD
6) Added sensitive detector ids for elements in the region after the neckdown (800-806)

rahmans1 committed:
Removing unused geometric elements

rahmans1 committed:
Changing dimension and extent of the pipe after detector pipe neckdown and before first flange in the region

rahmans1 committed:
Updating extents and dimension for the first flange before the bellows

rahmans1 committed:
Updating the extent and dimension of the bellows in the region

rahmans1 committed:
Fixed typo in variable name

rahmans1 committed:
Updating extent and dimension of the flange after the bellows in the region

rahmans1 committed:
Updating the dimension and extent of SAM pipe with end of the hall flange

rahmans1 committed:
Add an annulus of equivalent material to the SAM support around the pipe

rahmans1 committed:
Updating extent and dimension of detector pipe neckdown

rahmans1 committed:
Fixing typo: Extra =

rahmans1 committed:
Fix typo: Use appropriate end quotes for GDML

rahmans1 committed:
Adding position reference of subtracted volume

rahmans1 committed:
Updating neckdown reference frame to appropriately subtract the right volume

rahmans1 committed:
Update detector pipe dimensions and extents

rahmans1 committed:
Updating neckdown subtracted volume to allow proper subtraction

rahmans1 committed:
Extending mother volume to avoid overlap

rahmans1 committed:
Modifying mother volume of hall to avoid overlap with mother volume of updated beampipe

rahmans1 committed:
Subtracting a section of virtual hall det to avoid overlap with updated detector pipe

rahmans1 committed:
Adjusting the ranges of bellows flanges and aluminum beampipe flanges at junctions

rahmans1 committed:
Updating placeholder SAM positions to aligned with that in CAD

Updating placeholder SAM positions to aligned with that in CAD. Realistic SAM geometry will be implemented later.

rahmans1 committed:
Define new materials Inconel625 and Inconel625-Air mixture

Density for inconel625 is 1/(0.58/8.902+0.20/7.18+0.05/7.874+0.10/10.22+0.0307/8.57+0.010/16.654+0.001/2+0.005/7.44+0.005/2.33+0.00015/2.2+0.00015/2+0.004/2.699+0.004/4.54+0.01/8.9)=8.314 g/cm^3. Mass fractions taken from [here](https://www.specialmetals.com/documents/technical-bulletins/inconel/inconel-alloy-625.pdf) and densities taken from [here](https://www.fe.infn.it/u/paterno/Geant4_tutorial/slides_further/Geometry/G4_Nist_Materials.pdf). The density of inconel-air mixture is 1/(0.90/0.00120479+0.10/8.314)=0.001338 g/cm^3 assuming 10%-90% mixture.

rahmans1 committed:
Fix typo: material tag needs to be singular

rahmans1 committed:
Fix typo: G4_AIR is all capitalized

rahmans1 committed:
Update materials for bellows and bellows flanges

Set bellows material to Inconel625-Air mixture (10%-90%) and the flanges to Inconel625

rahmans1 committed:
Adding sensitive detector ids to elements in the SAM pipe region

rahmans1 committed:
Reduce mother volume radius to avoid overlap

cipriangal commented:
@rahmans1 I thought we didn't use the hall detectors in the actual geometry anymore. Could you remove them? (it will simplify the file and our lives). Once you do that let me know and I'll pull the geometry locally to have a closer look. From the PR here it looks great.

rahmans1 commented:
@cipriangal These are the three virtual detectors (one for the cylindrical wall and two as lids) that exist in hallDaugter_dump.gdml. Do you want me to take them all out? Should they be removed completely or migrated to mollerParallel?
![VirtualDet](https://user-images.githubusercontent.com/7409132/185942784-2f9b5b81-11a1-47bc-87ff-55408126a9d9.PNG)

cipriangal commented:
good points .. the round top can be removed (we never used it and it's overly complicated). The flat lid and the hall wall (which should be placed just shy of the wall itself) should be migrated to the parallel world (unless they are present there already). Thanks for checking this!

chandabindu commented:
Hi Sakib, 
I admit I haven't checked every detail.
But it looks okay visually. And it doesn't have any overlap issues.

the Al ring near the SAM inserts is ~57 mm thick and 177 mm long - Is it designed that way - it seems too much Al for me. Maybe it is required to support the SAMs.
The 'white' region near the hall end is Galactic. Could you place a cylinder with an approximate IR of SAM_Support_placeholder (=260.03 mm) of air to mimic the experimental condition? 
Thanks.

chandabindu reviewed (approved):
Please see the other comment.

rahmans1 committed:
Removing virtual detectors from hallDaughter_dump.gdml

The virtual detectors from hallDaughter_dump.gdml is being removed. The cylinder wall and flat lid will be migrated to moller parallel.

cipriangal commented:
@rahmans1 : once you put the detectors in the moller parallel world i'll merge this. Thanks!

rahmans1 committed:
Migrating hall virtual dets into moller parallel

rahmans1 committed:
Move hall flat lid detector to correct position

rahmans1 commented:
> Hi Sakib, I admit I haven't checked every detail. But it looks okay visually. And it doesn't have any overlap issues.
> 
> the Al ring near the SAM inserts is ~57 mm thick and 177 mm long - Is it designed that way - it seems too much Al for me. Maybe it is required to support the SAMs. The 'white' region near the hall end is Galactic. Could you place a cylinder with an approximate IR of SAM_Support_placeholder (=260.03 mm) of air to mimic the experimental condition? Thanks.

@chandabindu  The Al ring is a placeholder. It will be replaced later by realistic tubes that insert into the SAM pipe. Both the z and azimuthal extent is the worst case possible so that we can have conservative estimates for splashback. The area around the SAM support holder should be air and inside it should be 1e-2 Torr (should be the case all along the beamline but I wanted to get all elements aligned with CAD before I fix the air/vacuum boundaries in a separate PR).

rahmans1 commented:
> @rahmans1 : once you put the detectors in the moller parallel world i'll merge this. Thanks!

@cipriangal I have done the migration. Please check if they look right.

cipriangal commented:
everything looks good
